lib/src/platform_sliver_app_bar.dart

```diff
@@ -0,0 +1,297 @@
+      centerTitle: data?.centerTitle,
+      excludeHeaderSemantics: data?.excludeHeaderSemantics ?? false,
+      titleSpacing: data?.titleSpacing,
+      collapsedHeight: data?.collapsedHeight,
+      expandedHeight: data?.expandedHeight,
+      floating: data?.floating ?? false,
+      pinned: data?.pinned ?? false,
+      snap: data?.snap ?? false,
+      stretchTriggerOffset: data?.stretchTriggerOffset ?? 100.0,
+      onStretchTrigger: data?.onStretchTrigger,
+      shape: data?.shape,
+      toolbarHeight: data?.toolbarHeight ?? kToolbarHeight,
+      leadingWidth: data?.leadingWidth,
+      toolbarTextStyle: data?.toolbarTextStyle,
+      titleTextStyle: data?.titleTextStyle,
+      systemOverlayStyle: data?.systemOverlayStyle,
+      forceMaterialTransparency: data?.forceMaterialTransparency ?? false,
+      clipBehavior: data?.clipBehavior,
+    );
+  }
+
+  @override
+  CupertinoSliverNavigationBar createCupertinoWidget(BuildContext context) {
+    final data = cupertino?.call(context, platform(context));
+
+    return CupertinoSliverNavigationBar(
+      //Common
+      key: data?.widgetKey ?? widgetKey,
+      leading: data?.leading ?? leading,
+      automaticallyImplyLeading:
+          data?.automaticallyImplyLeading ?? automaticallyImplyLeading ?? true,
+      backgroundColor: data?.backgroundColor ?? backgroundColor,
+      stretch: data?.stretch ?? stretch ?? false,
+      largeTitle: data?.title ?? title,
+
+      //Cupertino only
+      automaticallyImplyTitle: data?.automaticallyImplyTitle ?? true,
+      alwaysShowMiddle: data?.alwaysShowMiddle ?? true,
+      previousPageTitle: data?.previousPageTitle,
+      middle: data?.middle,
+      trailing: data?.trailing,
+      border: data?.border ?? _kDefaultNavBarBorder,
+      brightness: data?.brightness,
+      padding: data?.padding,
+      transitionBetweenRoutes: data?.transitionBetweenRoutes ?? true,
+      heroTag: data?.heroTag ?? _defaultHeroTag,
+    );
+  }
+}
+
+//! Copied from file: /opt/homebrew/Caskroom/flutter/3.10.0/flutter/packages/flutter/lib/src/cupertino/nav_bar.dart
+const Color _kDefaultNavBarBorderColor = Color(0x4D000000);
+
+const Border _kDefaultNavBarBorder = Border(
+  bottom: BorderSide(
+    color: _kDefaultNavBarBorderColor,
+    width: 0.0, // 0.0 means one physical pixel
+  ),
+);
+
+const _HeroTag _defaultHeroTag = _HeroTag(null);
+
+@immutable
+class _HeroTag {
+  const _HeroTag(this.navigator);
+
+  final NavigatorState? navigator;
+
+  // Let the Hero tag be described in tree dumps.
+  @override
+  String toString() =>
+      'Default Hero tag for Cupertino navigation bars with navigator $navigator';
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) {
+      return true;
+    }
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
+    return other is _HeroTag && other.navigator == navigator;
+  }
+
+  @override
+  int get hashCode => identityHashCode(navigator);
+}
```

Comment:
Regarding these local variables, I didn't find a better solution, so I had to copy and paste them here intact.